### ISSUE TITLE
form: visualize collapsible sections that contain invalid info or errors

### ISSUE THREAD

**Issue by Kiarokh** [Lundalogik/lime-elements#3528]
## User story
Read more on https://product.lime-crm.com/client/object/feedback/6272/
> **Problem:** 
>There is no warning when creating a card with a required field that the specific field is the cause on why you can’t create the card. 
>
> **Today's only solution**
> Click into every field on the card and after that the warning icon will become active and detailing that the field is obligatory.
>
> **Impact**
> Confusion leading to them contacting support about something is wrong in lime.To give an example,one admin had made the cellphone field on the coworker card required and when another coworker tried to set a coworker card as inactive (made before the change) , it would not let them nor give them an explanation on why besides a star icon next to the cellphone field.

## Current behavior
A form can be rendered with collapsible sections, based on how the schema is structured. 

https://github.com/user-attachments/assets/db84ae48-eafb-42ee-957e-2d2628bb7ce0

A collapsible section can contain invalid fields. Individual fields (e.g. `limel-input-field`) nested inside the `limel-collapsible-section`s (which can have their own native validations or custom validations) can be invalid. 

This can happen either:
- when the form is loaded with invalid fields, or
- after some actions are done by the consumer, for example saving a form after the user has input some values in the form fields, can trigger a validation process. 
 
After validations are run, some form fields can then become invalid, for example those which are required but left empty by the user, or those which have an invalid value format.

In the current implementation of the `limel-form` component, the `limel-collapsible-section`s do not show any signs that they have one or more invalid fields inside. Therefore, if the collapsible section is closed, the user will have a really hard time understanding which collapsible section has the validation problem. 

## Expected behavior

Collapsible section have an `invalid` boolean prop, which can instead be set to true, if the section of the form contains invalid fields. Also, the invalid fields inside that section should turn "red" and clearly communicate that they are invalid, or required but empty.

https://github.com/user-attachments/assets/40cce3b9-789d-4acc-a307-2df19caefc31

**Comment by eketorp** [comment]
Can you explain how these sections will "turn" invalid in a form? In the Card form, Save becomes disabled when required fields are left empty. But I don't see how the user is able to "trigger" the check "is the form valid?"

Required fields can also become invalid if the user puts focus and the unfocuses the field. Is it that "if a section holds an invalid field, also make the section invalid"?

**Comment by Kiarokh** [comment]
Sure @eketorp I've added screen recordings to both "Current behavior" and "Expected behavior" sections, explaining the problem better, and also clarifying the proposed upcoming solution

**Comment by eketorp** [comment]
I see. My _personal_ impression of this is, just as I wrote in the comment in Product Lime:

> I am generally a bit skeptical to UI solutions that are of the sort “scream louder”, which “more highlighting” would fall under. I personally would find it annoying if a field screamed “hey, look at me!” and I would reply “hey, take it easy. i just started to fill out the form”. If all fields have been properly filled out before the user clicks save, no shouting will be necessary.

I am not saying that I have any better suggestion. And you are maybe not saying that this is the perfect design. But I'd love to hear what you think about "lighting up the form in red" as soon as the user starts editing it.

**Comment by Kiarokh** [comment]
Thanks for pushing back. I think you have a point there! 

The issue with the technical approach I've chosen, which is not visualized in the screen recording, is that all _required-but-empty-fields_ will start screaming in the same time, if the form is empty and the frist field is edited or filled in. This happens immediately after the first edit by the end user.  In reality, they might be intending to jump to the next required field and fill it in, but we start screaming before they even got the chance to do so.

I think I was too focused on the description of the problem, so that I forgot the scenario of "_What if the form is completely empty?_". So I had a tunnel vision.

The problem describes a scenario where the form is already pre-filled and just needs to be slightly edited. In this case, if there is a filed that is missing input or is invalid, then it will lit up. As user described, it can happen because the filed wasn't required a few days ago, but now is required, because an admin changed the props recently. 

So perhaps a combination of the two approaches could be good. The form field stays silent, until the user goes for the save. Then we somehow tell them that hey, there are empty fields or invalid fields in the form, they are in this particular section. Follow the red thread, and you'll find everything. 

I'm right now not sure whether this should happen when the user clicks the save button (which means the save button should remain `enabled` but actually dysfunctional), or that the save button should stay `disabled` and instead tell the user clearly why it is disabled. For example, by showing a popover, or dialog or something. 

I'll think a bit more, sketch a bit, and update the videos

**Comment by Kiarokh** [comment]
@eketorp updated the 2nd video. What do you think?
